sdk/ai/Azure.AI.Projects/README.md

```diff
@@ -210,6 +212,110 @@ Agent agent = agentResponse.Value;
+        dataSources: new List<VectorStoreDataSource> { ds }
+    )
+);
+var vectorStore = vectorStoreTask.Value;
+
+FileSearchToolResource fileSearchResource = new([vectorStore.Id], null);
+
+List<ToolDefinition> tools = [new FileSearchToolDefinition()];
+Response<Agent> agentResponse = await client.CreateAgentAsync(
+    model: modelName,
+    name: "my-assistant",
+    instructions: "You are helpful assistant.",
+    tools: tools,
+    toolResources: new ToolResources() { FileSearch = fileSearchResource }
+);
+```
+
+We also can attach files to the existing vector vector store. In the code snippet below, we first create an empty vector store and add file to it.
+
+```C# Snippet:BatchFileAttachment
+var ds = new VectorStoreDataSource(
+    assetIdentifier: blobURI,
+    assetType: VectorStoreDataSourceAssetType.UriAsset
+);
+var vectorStoreTask = await client.CreateVectorStoreAsync(
+    fileIds: null,
+    name: "sample_vector_store"
+);
+var vectorStore = vectorStoreTask.Value;
+
+var uploadTask = await client.CreateVectorStoreFileBatchAsync(
+    vectorStoreId: vectorStore.Id,
+    dataSources: new List<VectorStoreDataSource> { ds }
+);
+Console.WriteLine($"Created vector store file batch, vector store file batch ID: {uploadTask.Value.Id}");
+
+FileSearchToolResource fileSearchResource = new([vectorStore.Id], null);
+```
+
+#### Create Message with Code Interpreter Attachment
+
+To attach a file with the context to the message, the `MessageAttachment` class have to be used. To be able to process the attached file contents we need to provide the `List` with the single element `CodeInterpreterToolDefinition` as a `tools` parameter to both `CreateAgent` method and `MessageAttachment` class constructor.
+
+Here is an example to pass `CodeInterpreterTool` as tool:
+
+```C# Snippet:CreateAgentWithInterpreterTool
+AgentsClient client = new AgentsClient(connectionString, new DefaultAzureCredential());
+
+List<ToolDefinition> tools = [ new CodeInterpreterToolDefinition() ];
+Response<Agent> agentResponse = await client.CreateAgentAsync(
+    model: modelName,
+    name: "my-assistant",
+    instructions: "You are helpful assistant.",
+    tools: tools
+);
+Agent agent = agentResponse.Value;
+
+var fileResponse = await client.UploadFileAsync(filePath, AgentFilePurpose.Agents);
+var fileId = fileResponse.Value.Id;
+
+var attachment = new MessageAttachment(
+    fileId: fileId,
+    tools: tools
+);
+
+Response<AgentThread> threadResponse = await client.CreateThreadAsync();
+AgentThread thread = threadResponse.Value;
+
+Response<ThreadMessage> messageResponse = await client.CreateMessageAsync(
+    threadId: thread.Id,
+    role: MessageRole.User,
+    content: "What does the attachment say?",
+    attachments: new List< MessageAttachment > { attachment}
+    );
+ThreadMessage message = messageResponse.Value;
+```
+
+Azure blob storage can be used as a message attachment. In this case `VectorStoreDataSource` have to be used as a data source:
```

Comment:
Instead of 
   "In this case `VectorStoreDataSource` have to be used as a data source:"
I would write:
   "In this case, use `VectorStoreDataSource` as a data source:"

Reply:
Fixed.